[GPU Process][FormControls] Remove the flipping workaround for drawing a viewless NSCell

shallawa opened the pull request:
#### deb6370ae7025c39a7f2cc8263a04330872bd2e6
<pre>
[GPU Process][FormControls] Remove the flipping workaround for drawing a viewless NSCell
<a href="https://bugs.webkit.org/show_bug.cgi?id=252427">https://bugs.webkit.org/show_bug.cgi?id=252427</a>
rdar://105563614

Reviewed by Aditya Keerthi.

This patch will remove part of the workaround which was added in 260210@main.
But we still need to un-flip the flipped coordinates system before asking AppKit
to draw the NSCells. The only difference this patch makes is we will un-flip the
context unconditionally for all NSCell types.

* Source/WebCore/platform/cocoa/LocalCurrentGraphicsContext.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::drawSliderCell):
(WebCore::drawViewlessCell):
(WebCore::drawCellInView):
(WebCore::ControlMac::drawCellInternal):
(WebCore::ControlMac::drawCellFocusRingInternal):
(WebCore::ControlMac::drawCellFocusRing):
(WebCore::ControlMac::drawCellOrFocusRing):
(WebCore::shouldFlipContext): Deleted.
* Source/WebCore/platform/ios/LocalCurrentGraphicsContextIOS.mm:
(WebCore::LocalCurrentGraphicsContext::LocalCurrentGraphicsContext):
* Source/WebCore/platform/mac/LocalCurrentGraphicsContextMac.mm:
(WebCore::LocalCurrentGraphicsContext::LocalCurrentGraphicsContext):

Canonical link: <a href="https://commits.webkit.org/261085@main">https://commits.webkit.org/261085@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2571bad061c40c53a67a782609f457e14cc1de74

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110363 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19452 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42973 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1751 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119295 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114311 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20895 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10615 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102608 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116107 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98753 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43780 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97522 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30418 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85636 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12118 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31756 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12707 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8724 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18072 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51371 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7680 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14556 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
```diff
@@ -284,8 +282,10 @@ static void drawCellFocusRingInView(const FloatRect& rect, NSCell *cell, NSView
-void ControlMac::drawCellFocusRingInternal(const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell)
+void ControlMac::drawCellFocusRingInternal(GraphicsContext& context, const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell)
 {
+    LocalCurrentGraphicsContext localContext(context);
```
I'm assuming focus rings don't suffer from the flipping issue when used without a view?

shallawa replied:
We can't tell until rdar://105249508 is fixed. The focus ring is not displayed with viewless drawing. But also we can't create an NSView out of the main thread. This is why the focus ring is not working with GPUP for DOM rendering is on.

shallawa committed:
[GPU Process][FormControls] Remove the flipping workaround for drawing a viewless NSCell
https://bugs.webkit.org/show_bug.cgi?id=252427
rdar://105563614

Reviewed by Aditya Keerthi.

This patch will remove part of the workaround which was added in 260210@main.
But we still need to un-flip the flipped coordinates system before asking AppKit
to draw the NSCells. The only difference this patch makes is we will un-flip the
context unconditionally for all NSCell types.

* Source/WebCore/platform/cocoa/LocalCurrentGraphicsContext.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::drawSliderCell):
(WebCore::drawViewlessCell):
(WebCore::drawCellInView):
(WebCore::ControlMac::drawCellInternal):
(WebCore::ControlMac::drawCellFocusRingInternal):
(WebCore::ControlMac::drawCellFocusRing):
(WebCore::ControlMac::drawCellOrFocusRing):
(WebCore::shouldFlipContext): Deleted.
* Source/WebCore/platform/ios/LocalCurrentGraphicsContextIOS.mm:
(WebCore::LocalCurrentGraphicsContext::LocalCurrentGraphicsContext):
* Source/WebCore/platform/mac/LocalCurrentGraphicsContextMac.mm:
(WebCore::LocalCurrentGraphicsContext::LocalCurrentGraphicsContext):

Canonical link: https://commits.webkit.org/261085@main
